ext/tidy/tests/gh20374.phpt

```diff
@@ -0,0 +1,149 @@
+--TEST--
+GH-20374 (PHP with tidy and custom-tags)
+--EXTENSIONS--
+tidy
+--CREDITS--
+franck-paul
+--FILE--
+<?php
+
+class MyStringable {
+	public function __construct(private $ret) {}
+
+	public function __toString(): string {
+		return $this->ret;
+	}
+}
+
+class MyThrowingStringable {
+    public function __toString(): string {
+        throw new Error('no');
+    }
+}
+
+$values = [
+	'string blocklevel' => 'blocklevel',
+	'int' => 1,
+	'double overflow' => (string) (2.0**80.0),
+	'numeric string int 1' => '1',
+	'numeric string double 1.0' => '1.0',
```

Comment:
Can I be annoying and ask you to have test cases for INF and NAN? :)

Actually curious what the behaviour would be.

Reply:
Pushed. Apparently: nothing meaningful as they get turned into (int)0.